### PR TITLE
Wazo 3521 pydantic validation

### PR DESCRIPTION
**Pull request by sopelj**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ markupsafe==1.1.1 # from jinja
 pyopenssl>=21.0.0,<23.0.0  # bullseye uses 20.0.1, but only the patched debian version still works.
 cryptography>=3.3.2 # for pyopenssl
 pyyaml==5.3.1
+pydantic==1.7.4
```
Comment (fblackburn1) [inline requirements.txt:13]: don't forget to add pydantic to debian/control too

Comment (sopelj) [reply requirements.txt:13]: Ah oui, good point. Thanks.

**Comment by sopelj** [comment]
I can't seem to get the test_synchonize to pass. Somehow it is missing fields, although they seem to be present in the fixture used for the test. I'm not too certain where the config comes from though.

**Comment by sopelj** [comment]
Hey @bloom1, I got the validations I could find replaced with Schema Validations. Some I left to try and preserve the messages as discussed, but in theory they are not needed as that will be checked afterwards. I still have an issue with the test_synchonize method though, that I didn't have time to figure out.
I know the logic for the conversion of TypedDicts to pydantic models may not be super clear, so don't hesitate if you have questions. This is built into pydantic 1.9+ and unneeded in pydantic 2+ as you can directly use TypedDicts for validation.

**Review by bloom1 — APPROVED** [review]

**Pull request merged**